### PR TITLE
Check for signature for responses with empty body

### PR DESCRIPTION
**Pull request by itafroma**
The spec seems to indicate that X-Server-Authorization-HMAC-SHA256 should not be empty even if the response body is empty:

> The response signature base string is a concatenated string generated from the following parts:
> - `Nonce`:  The nonce that was sent in the Authorization header.
> - `Timestamp`: The timestamp that was sent in the X-Authorization-Timestamp header
> - `Body`: The response body (or empty string).

This adds the missing signature for the test fixtures.

### CODE REVIEW

**Comment by kevinhankens** [comment]
+1

**Comment by baliame** [comment]
There may be some similar holes in the tests - lacking a reference implementation at the time I only added what I did calculate myself step-by-step, thus the missing response signature. As we now consider this the reference implementation (I believe, at least), this does raise a very good point.

tl;dr: Good catch, thanks!
